modules/rosetta/src/test/scala/org/tessellation/rosetta/server/RosettaConstructionSuite.scala

```diff
@@ -0,0 +1,151 @@
+package org.tessellation.rosetta.server
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect.unsafe.implicits.global
+import cats.effect.{Async, IO, Resource}
+
+import scala.concurrent.duration.Duration
+import scala.io.Source
+
+import org.tessellation.ext.cats.effect.ResourceIO
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.rosetta.server.JsonObjects.RequestJson
+import org.tessellation.rosetta.server.JsonObjects.implicits._
+import org.tessellation.rosetta.server.runner.runnerKryoRegistrar
+import org.tessellation.sdk.config.types.{HttpClientConfig, HttpServerConfig}
+import org.tessellation.sdk.resources.MkHttpServer.ServerName
+import org.tessellation.sdk.resources.{MkHttpClient, MkHttpServer}
+import org.tessellation.security.SecurityProvider
+import org.tessellation.shared.sharedKryoRegistrar
+
+import com.comcast.ip4s.{Host, Port}
+import io.circe.parser.{decode, parse}
+import io.circe.syntax._
+import io.circe.{Json, JsonNumber}
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.Client
+import org.http4s.server.Server
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import suite.ResourceSuite
+import weaver.scalacheck.Checkers
+
+object RosettaConstructionSuite extends ResourceSuite with Checkers {
+  override type Res = (Resource[IO, Server], Resource[IO, Client[IO]])
+  override def sharedResource: Resource[IO, Res] = {
+    val server = serverResource()
+    val client = MkHttpClient
+      .forAsync[IO]
+      .newEmber(HttpClientConfig(Duration(5, TimeUnit.SECONDS), Duration(10, TimeUnit.SECONDS)))
+
+    IO.pure((server: Resource[IO, Server], client: Resource[IO, Client[IO]])).asResource
+  }
+
+  test("Construction test") { implicit resources: Res =>
+    val serverObj = resources._1
+    val httpClientObj = resources._2
+
+    serverObj.use { server =>
+      httpClientObj.use { client =>
+        val jsonString = Source.fromResource("rosetta.test_transactions.json").getLines().mkString
+        val cursor = parse(jsonString).getOrElse(Json.Null).hcursor
+        val params = cursor.downField("item").as[List[Json]].getOrElse(List.empty[Json])
+
+        val testResults = params.map { testEntry =>
+          val entryCursor = testEntry.hcursor
+          val payloadString =
+            entryCursor.downField("request").downField("body").downField("raw").as[String].getOrElse("{}")
+          val payload = decode[RequestJson](payloadString) match {
+            case Left(_)      => Json.Null
```

Comment:
Is this ignoring all the requests that don't conform to this schema? For instance `{{HOST}}:8080/construction/combine` has fields not present in this RequestJson schema. The response string appears to correspond to all the expected responses here but are we submitting an empty payload here?

Reply:
I am no longer using case classes to avoid this problem.